src/commands/Music/queue.ts

```diff
@@ -91,24 +96,53 @@ export default class extends MusicCommand {
 		return response.edit(undefined, queueDisplay.template);
 	}
 
-	private async generateSongField(message: KlasaMessage, position: number, song: Song) {
-		const username = await song.fetchRequesterName();
+	private async generateTrackField(message: GuildMessage, position: number, entry: DecodedQueueEntry) {
+		const username = await this.fetchRequesterName(message, entry.author);
 		return message.language.get(LanguageKeys.Commands.Music.QueueLine, {
 			position: position + 1,
-			duration: song.friendlyDuration,
-			title: song.safeTitle,
-			url: song.url,
+			duration: showSeconds(entry.data.length),
+			title: entry.data.title,
+			url: entry.data.uri,
 			requester: username
 		});
 	}
 
-	private calculateTotalDuration(songs: Song[]) {
+	private calculateTotalDuration(entries: readonly DecodedQueueEntry[]) {
 		let accumulator = 0;
-		for (const song of songs) {
-			if (song.stream) return -1;
-			accumulator += song.duration;
+		for (const entry of entries) {
+			if (entry.data.isStream) return -1;
+			accumulator += entry.data.length;
 		}
 
 		return accumulator;
 	}
+
+	private async fetchRequesterName(message: GuildMessage, userID: string): Promise<string> {
+		try {
+			return (await message.guild.members.fetch(userID)).displayName;
+		} catch {}
+
+		try {
+			return (await this.client.users.fetch(userID)).username;
+		} catch {}
+
+		return message.language.get(LanguageKeys.Misc.UnknownUser);
+	}
+
+	private async getTrackInformation(audio: Queue): Promise<readonly DecodedQueueEntry[]> {
+		const tracks = await audio.tracks();
+		const decodedTracks = await audio.player.node.decode(tracks.map((track) => track.track));
+
+		const map = new Map<string, TrackInfo>();
+		for (const entry of decodedTracks) {
+			map.set(entry.track, entry.info);
+		}
+
+		return tracks.map((track) => ({ author: track.author, data: map.get(track.track)! }));
+	}
+}
+
+interface DecodedQueueEntry {
```

Comment:
Why is this here?

Reply:
What do you mean? It's specific to the `queue` command.